pkg/git/client.go

```diff
@@ -132,45 +145,63 @@ func (c *client) Clone(ctx context.Context, repoID, remote, branch, destination
 		)
 	)
 
-	c.lockRepo(repoID)
-	defer c.unlockRepo(repoID)
-
-	_, err := os.Stat(repoCachePath)
-	if err != nil && !os.IsNotExist(err) {
-		return nil, err
+	authArgs := []string{}
+	if c.username != "" && c.password != "" {
+		token := fmt.Sprintf("%s:%s", c.username, c.password)
+		encodedToken := base64.StdEncoding.EncodeToString([]byte(token))
+		header := fmt.Sprintf("Authorization: Basic %s", encodedToken)
+		authArgs = append(authArgs, "-c", fmt.Sprintf("http.extraHeader=%s", header))
 	}
```

Comment:
📝 
The `authArgs` is used only in the `repoSingleFlights.Do`. so we can refactor and make these lines in the singleflight.Do's closure.
( I mistakenly put them out of closure. )

Reply:
📝 
I opened the PR for this.
https://github.com/pipe-cd/pipecd/pull/5190

Reply:
Let's add that PR to the cherrypick as well 👀

Reply:
I will recreate cherry-pick PR